src/browser.ts

```diff
@@ -137,14 +137,7 @@ class Honeybadger extends Client {
         // eslint-disable-next-line @typescript-eslint/no-this-alias
         const client = this
         func.___hb = <WrappedFunc>function () {
-          const onErrorEnabled = client.config.enableUncaught
-          // Catch if:
-          //   1. We explicitly want to catch (i.e. if the error could be
-          //      caught before reaching window.onerror)
-          //   2. The browser provides less information via the window.onerror
-          //      handler
-          //   3. The window.onerror handler is unavailable
-          if (opts.catch || preferCatch || !onErrorEnabled) {
+          if (preferCatch) {
```

Comment:
@subzero10 take a look at this. I think the `opts.catch` and `enableUncaught` checks were both unnecessary because the `window.onerror` is always available (even when `enableUncaught === false`). What we really want is just to choose which method to use to add the breadcrumb (should happens always) and report the error (should only happen when `enableUncaught === true`).

Reply:
The way I understand this is:
- sometimes `window.onerror` has more context
- with this block of code we were deciding whether we want to let a potential error be caught with the `window.onerror` listener or catch it here using try/catch

Based on the above:
- `opts.catch` was here to allow the user to take this decision, though I feel it may be unnecessary
- the `enableUncaught` looks redundant to me. The importance of this check is when calling `client.notify` which you added below (line 159)

In conclusion, I think your latest change makes the code more readable and covers the purpose of this issue/PR.

Reply:
Thanks @subzero10, I think you got it. I'm pretty sure the `enableUncaught` check also made more sense in the past because we didn't install the `window.onerror` handler when it was `false` (and it used to be called `onerror`), so in that case there would be no onerror handler and we'd have to always use `catch`.